Adapter.C

```diff
@@ -249,7 +256,10 @@ void preciceAdapter::Adapter::configure()
         DEBUG(adapterInfo("Creating interfaces..."));
         for (uint i = 0; i < interfacesConfig_.size(); i++)
         {
-            Interface* interface = new Interface(*precice_, mesh_, interfacesConfig_.at(i).meshName, interfacesConfig_.at(i).locationsType, interfacesConfig_.at(i).patchNames, interfacesConfig_.at(i).meshConnectivity);
+            std::string namePointDisplacement = FSIenabled_ ? FSI_->namePointDisplacement : "default";
+            std::string nameCellDisplacement = FSIenabled_ ? FSI_->nameCellDisplacement : "default";
+
+            Interface* interface = new Interface(*precice_, mesh_, interfacesConfig_.at(i).meshName, interfacesConfig_.at(i).locationsType, interfacesConfig_.at(i).patchNames, interfacesConfig_.at(i).meshConnectivity, interfacesConfig_.at(i).resetDisplacement, namePointDisplacement, nameCellDisplacement);
```

Comment:
Do we really need to pass these as arguments? We read them again in `FSI.C`, but I see that we now need them inside `Interface.C`. This starts getting too complicated (many of the changes in this PR are just for this), but I am not sure of any good solution at the moment.

I guess for now, we should either keep it as it is (but open an issue to fix the situation later), or keep the hard-coded names, documenting that restarting ignores these options. I think I would favor the hard-coded names for now (sorry for reverting your efforts).

What do you think?

Reply:
There are no hard-coded names here. If the user specifies names in the FSI part of the dict, they will be picked-up here. Am I missing something? The 'default' here is just to don't throw uninitialized memory around when the FSI module is disabled (and the option is irrelevant).

Reply:
Indeed, this part is not reading anything from the config (I somehow made this assumption while reviewing everything together), it is only accessing the now public variable, which has already been defined at the configuration of the FSI module.

Looking at it again, I think the main aspect I don't like is that we are passing even more parameters via constructor arguments. In the context of the current state (see also mesh connectivity), I guess this is fine.

Reply:
Yes, I was also briefly thinking about the constructor length while changing it here, but couldn't find a silver bullet solution right now.